### ISSUE TITLE
Integrate other structured generation backends

### ISSUE THREAD

**Issue by RobinPicard** [dottxt-ai/outlines#1654]
Outlines currently rely on `outlines-core` for creating the logits processor that does the actual logits biasing for local models. The idea would be to allow users to select alternative structured generation libraries to do that such as `xgrammar` or `llguidance`.

For this we will use a plugin system with a registry of backends.
